File: 2024_12_06_Python_Vs_Java_Challenges/Challenge_1/word.py
class Word:
    def __init__(self, word):
        self._word = word
        self._vowels = "aeiou"

    def count_vowels(self):
        count = 0
        for letter in self._word.lower():
            if letter in self._vowels:
                count += 1
        return count

    def count_consonants(self):
        return len(self._word) - self.count_vowels()

    def __str__(self):
        return self._word
    
if __name__ == "__main__":
    word = Word("string")
    print(word)
    print(word.count_vowels())
    print(word.count_consonants())

File: 2024_12_06_Python_Vs_Java_Challenges/Challenge_1/Word.java
public class Word {
    
    private String word;

    public Word(String word) {
        this.word = word;
    }

    public int countVowels() {
        int count = 0;
        for (int i = 0; i < word.length(); i++) {
            char letter = word.toLowerCase().charAt(i);
            if (letter == 'a' || letter == 'e' || letter == 'i' || letter == 'o' || letter == 'u') {
                count++;
            }
        }
        return count;
    }

    public int countConsonants() {
        return word.length() - countVowels();
    }

    public String toString() {
        return word;
    }
    
    public static void main(String[] args) {
        Word word = new Word("string");
        System.out.println(word);
        System.out.println(word.countVowels());
        System.out.println(word.countConsonants());
    }
}
